Enterprise-CMCS/macpro-mdct-mcr
# Refactor masking

### Description

Masking numerical user inputs in MCR has gotten complex, and requires a refactor to decrease code bloat + excessive reliance on masking over input validations.

More details [here](https://docs.google.com/document/d/1d0L2UyKcCU0lUJBFvGHJUJ-UIdredIasIHLO17a_wcM/edit)

### Related ticket(s)


MDCT-2597
MDCT-2665

---
### How to test

1. Login to MCR
2. Open either MCPAR or MLR, and navigate to a numerical input field 
3. Test inputs for validity (example valid/invalid inputs listed [here](https://docs.google.com/document/d/1d0L2UyKcCU0lUJBFvGHJUJ-UIdredIasIHLO17a_wcM/edit#heading=h.qgirlr2c8bqw))
4. If value is valid, confirm that it is getting masked correctly, and the cleaned value is getting sent to the database

### Important updates




---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
